packs/orion/actions/lib/actions.py

```diff
@@ -0,0 +1,108 @@
+# Licensed to the StackStorm, Inc ('StackStorm') under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import time
+
+from st2actions.runners.pythonrunner import Action
+from orionsdk import SwisClient
+
+
+class OrionBaseAction(Action):
+    def __init__(self, config):
+        super(OrionBaseAction, self).__init__(config)
+
+        self.swis = None
+
+        if "orion" not in self.config:
+            raise ValueError("Orion host details not in the config.yaml")
+
+    def connect(self, platform):
+        try:
+            self.swis = SwisClient(self.config['orion'][platform]['host'],
+                                   self.config['orion'][platform]['user'],
+                                   self.config['orion'][platform]['password'])
+        except KeyError:
+            raise ValueError("Orion host details not in the config.yaml")
+
+    def query(self, swql, **kargs):
+        return self.swis.query(swql, **kargs)
+
+    def invoke(self, entity, verb, *args):
+        return self.swis.invoke(entity, verb, *args)
+
+    def get_ncm_node_id(self, caption):
+        swql = "SELECT NodeID FROM Cirrus.Nodes WHERE NodeCaption=@node"
+        kargs = {'node': caption}
+        data = self.query(swql, **kargs)
+
+        if len(data['results']) == 1:
+            try:
+                return data['results'][0]['NodeID']
+            except IndexError:
+                raise IndexError("Invalid Node")
+        elif len(data['results']) >= 2:
+            raise IndexError("Muliple Nodes match '{}' NodeCaption".format(
+                caption))
+        elif len(data['results']) == 0:
+            raise IndexError("No matching NodeCaption for '{}'".format(
+                caption))
+
+    def get_ncm_transfer_results(self, transfer_id):
+        ts = {}
+        while True:
+            swql = """SELECT TransferID, Action, Status, ErrorMessage,
+            DeviceOutput FROM NCM.TransferResults
+            WHERE TransferID=@transfer_id"""
+            kargs = {'transfer_id': transfer_id}
+
+            transfer_data = self.query(swql, **kargs)
+            status = transfer_data['results'][0]['Status']
+
+            if status == 1:
+                time.sleep(10)
```

Comment:
I've make it configurable in the method, but I'm not sure how useful it will be to push that up to into the action.